#  Most of the parameters can be seen at the bottom of the page
#    https://rmcantin.bitbucket.io/html/usemanual.html#params
#  Please check the page if you want to know the parameters in detial, here I just list most important information of them.
rosbag_path: /home/zhaozhong/dataset/tum/tum.bag #develop for future slam_bayesopt in ros, ignore this
slam_yaml_path: /home/zhaozhong/VINS_catkin_ws/VINS-Mono/config/tum/tum_congfig.yaml # ignore this

#surrogate model. check https://rmcantin.bitbucket.io/html/bopttheory.html#surrmod for the choice surrogate model, kernel and acquisition function
surr_name: sStudentTProcessNIG 
kernel_name: kMaternARD5 
crit_name: cEI

n_init_samples: 20 #initial sample numbers
n_iterations: 50 #total iterations that need to be done
random_seed: -1 #random_seed. if it is smaller than 0, we choose random initial guess(time based). if it is equal or bigger than -1, we use fixed initial guess
verbose_level: 1 #control the output information. For instancem, if it is negative, we 'll output error, if it is 0, we'll output warning and error....
noise: 0.0000001 #1e-7 seems it does't accept this way. Noise of gaussian or student t process
force_jump: 10 #If bayesopt stucks at one point for "force_jump" times, we'll force it sample another points.
epsilon: 0.2 #not really used yet, see the first page

#If you choose optimizationChoice in vehicleParams yaml file as process noise, then opt_dim is 3. If you choose observation noise
#then opt_dim is 4. If you choose all, then it is 7. Lower bound should be bigger than zero because they are treat as the square of the noise.
opt_dim: 3 #dimension of the parameters that need to be optimized. must be at least 1 dimension of course
lower_bound: [0.00001, 0.00001, 0.00001] #lower bound, first one will be input as lowerbound(0), the second lowerbound(1).... dimension needs equal to opt_dim apparently
upper_bound: [0.2, 0.2, 0.2]

#kernel parameters
kernel_theta: 1.0 #for kernel theta and kernel sigma. See<Gaussian Process for Machine learning> chapter2.3
kernel_sigma: 10.0 #kernel theta is the l in chapter 2.3  
mean_name: mConst #mean of kernel, recommend to keep it as this
mean_mu: 1.0 #if set mean as constant, then mean == 1 or 0 is widely used. Thereotically constant mean won't influence the result
mean_sigma: 1000.0
prior_alpha: 1.0 #never used before
prior_beta: 1.0
sigma: 1.0

n_iteration_relearn: 50 #never modified before
n_inner_evaluation: 500 #never modified before
slam_fail_threshold: 99999999 #set by myself, for the use of SLAM initially. But it is not used now, just leave it.
ransac_noise: 0 #set by myself, for the use of SLAM initially. But it is not used now, just leave it.

#keep the following as it is
log_filename: bayesopt.log
save_filename: bayesopt.data
load_filename: bayesopt.data

